### PR TITLE
[IOTDB-912] Implement wasNull method in AbstractIoTDBJDBCResultSet

### PR DESCRIPTION
**Pull request by jack870131**
The wasNull method had been implemented and getXXX method should be modified according to the JDBC specification while there is null.

### CODE REVIEW

File: service-rpc/src/main/java/org/apache/iotdb/rpc/IoTDBRpcDataSet.java [hunk 1]
```diff
@@ -312,9 +317,23 @@ public float getFloat(String columnName) throws StatementExecutionException {
...
       return BytesUtils.bytesToFloat(values[index]);
     } else {
-      throw new StatementExecutionException(String.format(VALUE_IS_NULL, columnName));
+      lastReadWasNull = true;
+      return 0;
+    }
+  }
+
+  public short getShort(String columnName) throws StatementExecutionException {
+    checkRecord();
```
Comment (HTHou) [inline service-rpc/src/main/java/org/apache/iotdb/rpc/IoTDBRpcDataSet.java:329]: Actually, we are not supporting short type of data yet...

Comment (jack870131) [reply service-rpc/src/main/java/org/apache/iotdb/rpc/IoTDBRpcDataSet.java:329]: I think so, too. I felt weird that why it was implemented. How about just throwing exception while it has been used?

Comment (HTHou) [reply service-rpc/src/main/java/org/apache/iotdb/rpc/IoTDBRpcDataSet.java]: Yes, look good to me. 👍

**Review by HTHou — APPROVED** [review]

File: service-rpc/src/main/java/org/apache/iotdb/rpc/IoTDBRpcDataSet.java [hunk 2]
```diff
@@ -343,9 +352,11 @@ public long getLong(String columnName) throws StatementExecutionException {
     }
     int index = columnOrdinalMap.get(columnName) - START_INDEX;
     if (!isNull(index, rowsIndex - 1)) {
+      lastReadWasNull = false;
       return BytesUtils.bytesToLong(values[index]);
     } else {
-      throw new StatementExecutionException(String.format(VALUE_IS_NULL, columnName));
```
Comment (qiaojialin) [inline service-rpc/src/main/java/org/apache/iotdb/rpc/IoTDBRpcDataSet.java:348]: Maybe throw an exception is better? which can help users find bugs in client

Comment (jack870131) [reply service-rpc/src/main/java/org/apache/iotdb/rpc/IoTDBRpcDataSet.java:348]: The reason why I remove the exception here is that it was not consistent with JDBC specification as we discussed in [IOTDB-912](https://issues.apache.org/jira/projects/IOTDB/issues/IOTDB-912?filter=allopenissues).

Comment (qiaojialin) [reply service-rpc/src/main/java/org/apache/iotdb/rpc/IoTDBRpcDataSet.java:348]: ok thanks!

**Pull request merged**